### PR TITLE
Added a new oauth test client written in clojure

### PR DESCRIPTION
**Pull request by drakezhard**

### CODE REVIEW

**Comment by alangpierce** [comment]
Sounds cool, although I can't seem to find the test client itself. Looks like you added it as a submodule, but `git submodule init` is saying this:
```
No submodule mapping found in .gitmodules for path 'examples/clojure_test_client'
```
Maybe you accidentally referenced a non-public repo, or made some mistake in adding the submodule? Or I may be missing some detail about how to work with submodules.

I think it probably makes sense to add sample clients directly rather than as submodules.

**Comment by drakezhard** [comment]
This is my first pull request ever, I just copied my working repo and didn't delete the .git directory sorry, it's fixed now.

File: examples/clojure_test_client/src/khan_client/web.clj [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+(def credentials (atom {}))
+
+(defroutes app
+  (GET "/" []
+       (khan-api))
+  (GET "/auth" {params :params}
+       (swap! access-token-response conj (khan/access-token-request params))
+       (swap! credentials conj (khan/credential-handler @access-token-response 
+                                                        {:email "suma_coach@seatacunited.org"}))
+       (http/get "https://www.khanacademy.org/api/v1/user?email=suma_coach@seatacunited.org"
```
Comment (alangpierce) [inline examples/clojure_test_client/src/khan_client/web.clj]: Looks like you have your email address here. Can you change it to be more generic? I think `/api/v1/user` defaults to the current user, so you could just leave it out. If you want to demonstrate how to pass parameters, you could pass `format=pretty` or use some other endpoint.

**Comment by alangpierce** [comment]
Thanks, looks pretty reasonable to me (although I haven't worked with Clojure before). Sorry for the delay in getting to this.

I left one comment that I'd like to see addressed before I merge this. Also, it looks like you have your API key in one of the intermediate commits in the pull request. Can you squash the commits (using interactive rebase) to get rid of that?

**Pull request closed without merging**